protobuf-serializer/src/main/java/io/confluent/kafka/serializers/protobuf/AbstractKafkaProtobufDeserializer.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+
+package io.confluent.kafka.serializers.protobuf;
+
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.MessageLite;
+import kafka.utils.VerifiableProperties;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.errors.SerializationException;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchemaProvider;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchemaUtils;
+import io.confluent.kafka.serializers.AbstractKafkaSchemaSerDe;
+
+public abstract class AbstractKafkaProtobufDeserializer<T extends MessageLite>
+    extends AbstractKafkaSchemaSerDe {
+  protected Class<T> specificProtobufClass;
+  protected Method parseMethod;
+
+  /**
+   * Sets properties for this deserializer without overriding the schema registry client itself.
+   * Useful for testing, where a mock client is injected.
+   */
+  @SuppressWarnings("unchecked")
+  protected void configure(KafkaProtobufDeserializerConfig config) {
+    configureClientProperties(config, new ProtobufSchemaProvider());
+    try {
+      this.specificProtobufClass = (Class<T>) config.getClass(
+          KafkaProtobufDeserializerConfig.SPECIFIC_PROTOBUF_CLASS_CONFIG);
+      if (specificProtobufClass != null && !specificProtobufClass.equals(Object.class)) {
+        this.parseMethod = specificProtobufClass.getDeclaredMethod("parseFrom", ByteBuffer.class);
+      }
+    } catch (Exception e) {
+      throw new ConfigException("Proto class "
+          + specificProtobufClass.getCanonicalName()
+          + " is not a valid proto3 message class", e);
+    }
+  }
+
+  protected KafkaProtobufDeserializerConfig deserializerConfig(Map<String, ?> props) {
+    try {
+      return new KafkaProtobufDeserializerConfig(props);
+    } catch (io.confluent.common.config.ConfigException e) {
+      throw new ConfigException(e.getMessage());
+    }
+  }
+
+  protected KafkaProtobufDeserializerConfig deserializerConfig(VerifiableProperties props) {
+    return new KafkaProtobufDeserializerConfig(props.props());
+  }
+
+  private ByteBuffer getByteBuffer(byte[] payload) {
```

Comment:
I feel like this method could be placed in a more centralized location, if it doesn't already exist in some fashion

Reply:
Ack